File: exam-system-java/entity/src/main/java/com/yyjy/exam/entity/paper/entity/Paper.java
package com.yyjy.exam.entity.paper.entity;

import com.yyjy.exam.entity.user.entity.Users;
import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.time.LocalDateTime;
import java.util.List;

@Entity
public interface Paper {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	int id();
	
	@ManyToOne
	Users user();
	
	@JoinColumn(name = "category_id")
	@ManyToOne
	PaperCategories category();
	
	@IdView
	long categoryId();
	
	String name();
	
	@Nullable
	String description();
	
	@Column(name = "is_published")
	boolean published();
	
	@Nullable
	Double totalScore();
	
	@Nullable
	Double PassingScore();
	
	@Nullable
	Integer questionCount();
	
	int duration();
	
	LocalDateTime start();
	
	LocalDateTime end();
	
	@OneToMany(mappedBy = "paper")
	List<PaperQuestion> paperQuestions();
	
	@Nullable
	LocalDateTime createTime();
	
	@Nullable
	LocalDateTime updateTime();
	
	@Column(name = "deleted_at")
	@LogicalDeleted("now")
	@Nullable
	LocalDateTime delFlag();
}

File: exam-system-server/src/main/kotlin/com/yyjy/models/entity/Paper.kt
package com.yyjy.models.entity

import org.babyfish.jimmer.sql.*
import java.time.LocalDateTime

/**
 * 试卷表（试卷基本信息）
 */
@Entity
interface Paper {

    @Id
    @GeneratedValue(
        strategy = GenerationType.IDENTITY
    )
    val id: Int

    val name: String

    val description: String?

    /**
     * 试卷状态: DRAFT（草稿）/PUBLISHED（已发布）/STOPPED（已停用）
     */
    val status: String

    /**
     * 总分
     */
    val totalScore: Double?

    val questionCount: Int?

    val duration: Int

//    @ManyToManyView(
//        prop = "paperQuestions",
//        deeperProp = "question"
//    )
//    val questions: List<Questions>

    @OneToMany(mappedBy = "paper")
    val paperQuestions: List<PaperQuestion>

    /**
     * 创建时间
     */
    val createTime: LocalDateTime?

    /**
     * 更新时间
     */
    val updateTime: LocalDateTime?

    /**
     * 删除时间
     */
    @Column(name = "deleted_at")
    @LogicalDeleted("now")
    val delFlag: LocalDateTime?
}

